### PR TITLE
feat: implement finance page and api Endpoint

### PR DESCRIPTION
**Pull request by Esskla**
Added a view for an overview of the finances history. By default the last and current month will be displayed separately.
Also added a daterange picker to get the finances of multiple days/weeks or months.

The API and button for the export functions are also implemented. Only the Backend function and PDF generation is missing.

### CODE REVIEW

File: src/Resources/src/api/getFinances.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import useApi from "./api";
+import {Finances} from "@/stores/financesStore";
+import moment from "moment/moment";
+
+export default async function getFinances(dateRange?: Date[]) {
+  let requestUrl = 'api/accounting/book/finance/list'
+
+  if (dateRange !== undefined) {
+    const minDate = moment(dateRange[0]).format("YYYY-MM-DD");
```
Comment (hacksch) [inline src/Resources/src/api/getFinances.ts:9]: use single quotes

File: src/Resources/src/components/finance/FinanceHeader.vue [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import {onBeforeMount, ref} from "vue";
+import {useI18n} from "vue-i18n";
+import { DownloadIcon } from '@heroicons/vue/outline'
+import VueDatePicker from '@vuepic/vue-datepicker';
+import moment from "moment/moment";
+
+const emit = defineEmits(['dateChanged'])
+const { t, locale } = useI18n()
+const date = ref()
+let minDate = "";
```
Comment (hacksch) [inline src/Resources/src/components/finance/FinanceHeader.vue:47]: use single quotes

File: src/Resources/src/api/getFinances.ts (hunk 1 above)
Comment (Esskla) [reply src/Resources/src/api/getFinances.ts]: done :)

File: src/Resources/src/components/finance/FinanceHeader.vue (hunk 2 above)
Comment (Esskla) [reply src/Resources/src/components/finance/FinanceHeader.vue]: done :)

File: src/Resources/src/stores/financesStore.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        message: 'finance.empty'
+      })
+    } else {
+      FinancesState.finances = [];
+      FinancesState.error = 'Error on fetching finances';
+    }
+
+    FinancesState.isLoading = false;
+  }
+
```
Comment (hacksch) [inline src/Resources/src/stores/financesStore.ts:102]: leerzeile zuviel

**Review by hacksch — APPROVED** [review]

**Pull request merged**